### PR TITLE
Allow eligible status in ECFParticipantEligiblilty

### PR DESCRIPTION
**Pull request by tonyheadford**
## Ticket and context

Enable the `:eligible` status to be set in `ECFParticipantEligibility` instead of forcing the status to `:matched`. Ensure that this is still handled in the validations controller as though it were `:matched` status. This is to enable us to get a picture of who will successfully validate and the different outcomes that will need to be handled.

## Tech review
When we have matched the participants TRN, the qts flag is set and there are no other flags, the eligibility status should be `:eligible`.  In this instance we should delete/not store the validation data, however as this is not complete yet, we need to hang on to the personal data needed to re-run validation against the participant once it is complete.

### Code quality checks
- [ ] All commit messages are meaningful and true
- [ ] Added enough automated tests

### HTML Checks
- [ ] All new pages have automated accessibility checks
- [ ] All new pages have visual tests via Percy

### Gotchas
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Follow the steps from the ticket.

## External API changes

Does this change anything in our external APIs? If so, did you remember to update the CHANGELOG for Lead Providers? (lead-providers/guidance/release-notes)

### CODE REVIEW

**Review by kicferk1 — COMMENTED** [review]
Sweet!

Percy changes look a bit surprising - one of them seems spurious, and we need to fix it. The other seems a bit weirder, perhaps some conditional changes what text is being displayed on a dashboard. Could you double check that?

**Comment by tonyheadford** [comment]
> Sweet!
> 
> Percy changes look a bit surprising - one of them seems spurious, and we need to fix it. The other seems a bit weirder, perhaps some conditional changes what text is being displayed on a dashboard. Could you double check that?

Yes, just seen that, will investigate

File: app/controllers/participants/validations_controller.rb [hunk 1]
```diff
@@ -98,7 +98,8 @@ def cannot_find_details
 
     def complete
       @school = participant_profile.school
-      if participant_profile.ecf_participant_eligibility&.matched_status?
+      if participant_profile.ecf_participant_eligibility&.eligible_status? ||
+          participant_profile.ecf_participant_eligibility&.matched_status?
         # Probably successful - show success message.
         # TODO CPDRP-884: change line above to be eligible_status?
```
Comment (jamesrose) [inline app/controllers/participants/validations_controller.rb:104]:
```suggestion
```

File: spec/services/validate_participant_spec.rb [hunk 2]
```diff
@@ -118,7 +118,7 @@
             .and_return(eligible_record)
         end
 
-        it "does not save the validation data" do
+        xit "does not save the validation data" do
```
Comment (kicferk1) [inline spec/services/validate_participant_spec.rb:121]: Is that going back in when Phil's PR with new dqt api goes in? https://github.com/DFE-Digital/early-careers-framework/pull/1059

**Review by kicferk1 — APPROVED** [review]

File: spec/services/validate_participant_spec.rb (hunk 2 above)
Comment (tonyheadford) [reply spec/services/validate_participant_spec.rb:121]: Yes, once we can access the previous inductees we won't keep the validation data

**Comment by ethanmills** [comment]
I believe this is surfaced in the API. Should it be?

**Comment by kicferk1** [comment]
> I believe this is surfaced in the API. Should it be?

I don't see why not. The providers are interested in whether their participants are eligible for payment, and their verification status is the only thing that makes a difference in it at the moment. So it feels correct to surface that.

**Comment by ethanmills** [comment]
> > I believe this is surfaced in the API. Should it be?
> 
> I don't see why not. The providers are interested in whether their participants are eligible for payment, and their verification status is the only thing that makes a difference in it at the moment. So it feels correct to surface that.

The conversations I heard were that we weren't ready to share this data yet (hence not updating the UI)

**Comment by kicferk1** [comment]
> The conversations I heard were that we weren't ready to share this data yet (hence not updating the UI). Quite possible that something changed though

Uhh, if there is a very good reason not to share it, I would be interested to hear it. But in general I would be very keen to get it moving.

I guess we could be concerned about false negatives, until we get the full api merged, but they don't sound like a massive risk - at the moment the api says nobody is eligible for payment, which surely means more false negatives.

**Comment by tonyheadford** [comment]
> > > I believe this is surfaced in the API. Should it be?
> > 
> > 
> > I don't see why not. The providers are interested in whether their participants are eligible for payment, and their verification status is the only thing that makes a difference in it at the moment. So it feels correct to surface that.
> 
> The conversations I heard were that we weren't ready to share this data yet (hence not updating the UI). Quite possible that something changed though

I think you might be right @ethanmills, we don't want to make it public yet, we're not letting the participants know and we haven't got the previous inductees from the API yet (assuming that will work).  It might be better to hide this in the API for now

**Review by ethanmills — APPROVED** [review]
LGTM!

**Pull request merged**